shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/stopDetailsPage/TripHeaderSpec.kt

```diff
@@ -0,0 +1,48 @@
+package com.mbta.tid.mbta_app.model.stopDetailsPage
+
+import com.mbta.tid.mbta_app.model.Stop
+import com.mbta.tid.mbta_app.model.TripDetailsStopList
+import com.mbta.tid.mbta_app.model.Vehicle
+
+sealed class TripHeaderSpec {
+    data object FinishingAnotherTrip : TripHeaderSpec()
+
+    data object NoVehicle : TripHeaderSpec()
+
+    data class Scheduled(val stop: Stop, val entry: TripDetailsStopList.Entry) : TripHeaderSpec()
+
+    data class VehicleOnTrip(
+        val vehicle: Vehicle,
+        val stop: Stop,
+        val entry: TripDetailsStopList.Entry?
+    ) : TripHeaderSpec()
+
+    companion object {
+        fun getSpec(
```

Comment:
Pulled from [TripDetailsView.swift](https://github.com/mbta/mobile_app/blob/e8e1fe6c26e3f44424b3c21e575d1d1e7de3e8bf/iosApp/iosApp/Pages/StopDetails/TripDetailsView.swift#L122). Could probably safely convert that to use this, but punting on that for now.